server/src/main/java/io/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -361,41 +385,67 @@ private void loadAllLookupsAndInitStateRef()
   private List<LookupBean> getLookupListFromCoordinator(String tier)
   {
     try {
-      final FullResponseHolder response = fetchLookupsForTier(tier);
-      List<LookupBean> lookupBeanList = new ArrayList<>();
-      if (!response.getStatus().equals(HttpResponseStatus.OK)) {
-        LOG.error(
-            "Error while fetching lookup code from Coordinator with status[%s] and content[%s]",
-            response.getStatus(),
-            response.getContent()
-        );
-        return null;
-      }
-
-      // Older version of getSpecificTier returns a list of lookup names.
-      // Lookup loading is performed via snapshot if older version is present.
-      // This check is only for backward compatibility and should be removed in a future release
-      if (response.getContent().startsWith("[")) {
-        LOG.info("Failed to retrieve lookup information from coordinator. Attempting to load lookups using snapshot instead");
-        return null;
-      } else {
-        Map<String, LookupExtractorFactoryContainer> lookupMap = jsonMapper.readValue(
-            response.getContent(),
-            LOOKUPS_ALL_REFERENCE
-        );
+      MutableBoolean firstAttempt = new MutableBoolean(true);
+      Map<String, LookupExtractorFactoryContainer> lookupMap = RetryUtils.retry(
+          () -> {
+            if (firstAttempt.isTrue()) {
+              firstAttempt.setValue(false);
+            } else {
+              // Adding an extra minute in addition to the retry wait. In RetryUtils, retry wait starts from a few
+              // seconds, that is likely not enough to coordinator to be back to healthy state, e. g. if it experiences
+              // 30-second GC pause.
+              Thread.sleep(60_000);
+            }
+            return tryGetLookupListFromCoordinator(tier);
+          },
+          e -> true,
+          lookupConfig.getCoordinatorFetchRetries()
+      );
+      if (lookupMap != null) {
+        List<LookupBean> lookupBeanList = new ArrayList<>();
         lookupMap.forEach((k, v) -> lookupBeanList.add(new LookupBean(k, null, v)));
-
+        return lookupBeanList;
+      } else {
+        return null;
       }
-      return lookupBeanList;
     }
     catch (Exception e) {
       LOG.error(e, "Error while trying to get lookup list from coordinator for tier[%s]", tier);
       return null;
     }
   }
 
+  @Nullable
+  private Map<String, LookupExtractorFactoryContainer> tryGetLookupListFromCoordinator(String tier) throws Exception
+  {
+    final FullResponseHolder response = fetchLookupsForTier(tier);
+    if (!response.getStatus().equals(HttpResponseStatus.OK)) {
+      LOG.error(
```

Comment:
If this can retry it is not an error. Suggest just throwing a RE and letting the retry wrapper handle the logging

Reply:
Throwing IOE